encoders/src/test/java/au/csiro/pathling/encoders2/ResourceEncodingTest2.java

```diff
@@ -0,0 +1,96 @@
+package au.csiro.pathling.encoders2;
+
+import static org.junit.Assert.assertTrue;
+
+import au.csiro.pathling.encoders.EncoderUtils;
+import au.csiro.pathling.encoders.FhirEncoders;
+import au.csiro.pathling.encoders.datatypes.R4DataTypeMappings;
+import ca.uhn.fhir.context.FhirContext;
+import ca.uhn.fhir.context.RuntimeResourceDefinition;
+import com.google.common.collect.ImmutableSet;
+import java.util.Collection;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.spark.sql.catalyst.InternalRow;
+import org.apache.spark.sql.catalyst.encoders.ExpressionEncoder;
+import org.apache.spark.sql.catalyst.encoders.ExpressionEncoder.Serializer;
+import org.apache.spark.sql.types.StructType;
+import org.hl7.fhir.instance.model.api.IBaseResource;
+import org.hl7.fhir.r4.model.Base;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+
+@RunWith(Parameterized.class)
+public class ResourceEncodingTest2 {
+
+  private final static FhirContext fhirContext = FhirContext.forR4();
+  private final static FhirEncoders fhirEncoders = FhirEncoders.forR4().getOrCreate();
+
+
+  // TODO: Remove when the corresponding issues are fixed (#375)
+  private final static Set<String> EXCLUDED_RESOURCES = ImmutableSet.of(
+      "Parameters",
+      // Collections are not supported for custom encoders for: condition-> RuntimeIdDatatypeDefinition[id, IdType]
+      "Task",
+      // Collections are not supported for custom encoders for: condition-> RuntimeIdDatatypeDefinition[id, IdType]
+      "StructureDefinition",
+      // Collections are not supported for custom encoders for: condition-> RuntimeIdDatatypeDefinition[id, IdType]
+      "StructureMap",
+      // Collections are not supported for custom encoders for: condition-> RuntimeIdDatatypeDefinition[id, IdType]
+      "Bundle"
+      // scala.MatchError: RuntimeElementDirectResource[DirectChildResource, IBaseResource] (of class ca.uhn.fhir.context.RuntimeElementDirectResource)
+  );
```

Comment:
I think we should just explicitly throw an error when the user tries to import these types of resources.

Reply:
I have added a check for unsupported resources in EncoderBuilder2.